burnman/seismic.py

```diff
@@ -14,38 +20,31 @@ def __init__(self):
 
 
 
-    def evaluate_all_at(self, depth_list):
+    
+    def evaluate(self, vars_list, depth_list):
         """
         Returns the lists of data for a Seismic1DModel for the depths provided
-            
+        
         Parameters
         ----------
+        vars_list : array of str
+	    Variables to be evaluated. Options are depending on the model 'pressure','density','gravity','v_s','v_p','v_phi','G','K','QG','QK'
         depth_list : array of floats
             Array of depths [m] to evaluate seismic model at.
-            
+        
         Returns
         -------
-        pressures : array of floats
-            Pressures at given depths in [Pa].
-        density :  array of floats
-            Densities at given depths in [kg/m^3].
-        v_p : array of floats
-            P wave velocities at given depths in [m/s].
-        v_s : array of floats
-            S wave velocities at given depths in [m/s].
-        v_phi : array of floats
-            Bulk sound velocities at given depths in [m/s].
+        Array of values shapes as (len(vars_list),len(depth_list)). 
+
         """
-        pressures = np.array([self.pressure(r) for r in depth_list])
-        density = np.array([self.density(r) for r in depth_list])
-        v_p = np.array([self.v_p(r) for r in depth_list])
-        v_s = np.array([self.v_s(r) for r in depth_list])
-        v_phi = np.array([self.v_phi(r) for r in depth_list])
-        return pressures, density, v_p, v_s, v_phi
-    
+        
+        values=np.empty((len(vars_list),len(depth_list)))
+        for a in range(len(vars_list)):
+            values[a,:]=getattr(self,vars_list[a])(depth_list)
```

Comment:
I like this change. Perhaps it would be a good idea to check whether a seismic model has this particular variable implemented? Of course, if it does not then an exception will be raised, so maybe that is enough.